lib/knex/query.ts

```diff
@@ -238,7 +239,16 @@ module.exports = function (PersistObjectTemplate) {
                 var type = defineProperty.type;
                 var of = defineProperty.of;
                 var cascadeFetch = (cascade && typeof(cascade[prop] != 'undefined')) ? cascade[prop] : null;
-
+                if (cascadeFetch && cascadeFetch.fetch) {
```

Comment:
is there any limit on how deep we let the recursion go? e.g. if this recurses 100 times are we ok with that? what does that query end up looking like?

Reply:
is there any way somebody could accidentally blow things up by setting a recursive fetch spec on an object with a circular reference?

Reply:
I had the same concern as nick here, how do we limit this incase of gigantic fetches?

Reply:
I've created https://massmutual.atlassian.net/browse/HL-16058..
Can I get approvals on this PR